frontend/src/metabase/collections/components/PinnedQuestionCard/PinnedQuestionLoader.tsx

```diff
@@ -41,11 +41,10 @@ const PinnedQuestionLoader = ({
 
   return (
     <Questions.Loader id={id} loadingAndErrorWrapper={false}>
-      {({ loading, question: loadedQuestion }: QuestionLoaderProps) => {
-        if (loading || !loadedQuestion.query()) {
```

Comment:
The first render has `loading: undefined` and `loadedQuestion` presented. I guess `loadedQuestion` is always presented because it's cached from a collection list.

So the issue was we:
1. render `QuestionResultLoader` while `loading: undefined`
2. render this block when `loading: true`
3. render `QuestionResultLoader` again after `loading: false`

Reply:
This changes the initial [fix attempt](https://github.com/metabase/metabase/pull/35346/commits/e4d2269ee52fdc87bc9f6ea1afc43a70867cbc4f). Since it failed 1 E2E test on [displaying pivot table.](https://www.deploysentinel.com/ci/runs/6544a00a7bcc148681654fff)

The reason is that the `loadedQuestion` that's cached from `GET /api/collection/root/items` doesn't have the same `query` as the one loaded from `GET /api/card/:id`, that's why I still need to have this block that will render another loader before rendering `QuestionResultLoader`